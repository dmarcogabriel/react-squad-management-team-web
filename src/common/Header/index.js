import React from 'react';

const Header = () => (
  <nav
    data-testid="header"
    className="flex px-8 py-2 items-center justify-between bg-gradient-to-r from-primary-dark to-secondary-dark text-white"
  >
    <div>
      <p className="text-lg">Squad Management Tool</p>
    </div>

    <div className="flex items-center">
      <p className="text-sm mr-2 invisible md:visible">John Doe</p>

      <p className="bg-white py-2 px-3 rounded-full text-black">JD</p>
    </div>
  </nav>
);

export default Header;
