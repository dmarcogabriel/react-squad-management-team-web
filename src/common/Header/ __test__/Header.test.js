import { render } from '@testing-library/react';
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Header from '..';

describe('<Header />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Header />, div);
  });

  it('renders classes correctly', () => {
    const { getByTestId } = render(<Header />);

    expect(getByTestId('header')).toHaveClass(
      'flex px-8 py-2 items-center justify-between bg-gradient-to-r from-primary-dark to-secondary-dark text-white'
    );
  });

  it('matches snapshot', () => {
    expect(renderer.create(<Header />).toJSON()).toMatchSnapshot();
  });
});
