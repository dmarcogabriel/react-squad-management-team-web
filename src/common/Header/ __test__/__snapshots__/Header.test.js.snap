// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> matches snapshot 1`] = `
<nav
  className="flex px-8 py-2 items-center justify-between bg-gradient-to-r from-primary-dark to-secondary-dark text-white"
  data-testid="header"
>
  <div>
    <p
      className="text-lg"
    >
      Squad Management Tool
    </p>
  </div>
  <div
    className="flex items-center"
  >
    <p
      className="text-sm mr-2 invisible md:visible"
    >
      John Doe
    </p>
    <p
      className="bg-white py-2 px-3 rounded-full text-black"
    >
      JD
    </p>
  </div>
</nav>
`;
